src/libraries/System.Linq/tests/EmptyPartitionTests.cs

```diff
@@ -29,15 +29,15 @@ public void SingleInstance()
             Assert.True(ReferenceEquals(GetEmptyPartition<int>(), GetEmptyPartition<int>()));
         }
 
-        [Fact]
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsSpeedOptimized))]
```

Comment:
Is this behavior still something we can preserve even if `IsSizeOptimized`?

If not, it would be nice to add comments on why these tests are skipped. Reading the comment below makes it sound like these tests should work on all .NET Core builds.

Reply:
> Is this behavior still something we can preserve even if IsSizeOptimized?

Not without adding more code back to the size-optimized implementation. The comment below assumes the optimizations in the .Speed.cs implementation, which recognizes "partitions" and thus know that Skip(2) on an empty partition is also empty.  The Size implementation loses all such optimizations.